File: dbScripts/model/Review.py
from utilities import *


class Review:
    def __init__(self, id, customer_id, product_id, reviewText, createdAt, numberLikes):
        self._id = id
        self._customer_id = customer_id
        self._product_id = product_id
        self._reviewText = reviewText
        self._created_at = createdAt
        self._numberLikes = numberLikes

    def __str__(self) -> str:
        return f"({self._id}," + \
               f"{self._customer_id}," + \
               f"{self._product_id}," + \
               f"{get_string_in_single_quotes(self._reviewText)}," + \
               f"{get_string_in_single_quotes(self._created_at)}," + \
               f"{self._numberLikes})"

File: src/main/java/com/example/A2/domain/Review.java
package com.example.A2.domain;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.*;

@Entity
@Table(name="review")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Review {
    private @Id @GeneratedValue(strategy = GenerationType.IDENTITY) Long id;

    @ManyToOne
    @JoinColumn(name = "customer_id")
    @NotNull
    private Customer customer;

    @ManyToOne
    @JoinColumn(name = "product_id")
    @NotNull
    private Product product;

    @NotBlank(message = "Please enter the review")
    @Size(min=4, message = "The review should be at least 4 characters long")
    private String reviewText;

    private String createdAt;

    @Column(name = "numberLikes")
    private Integer numberLikes;

}
